### PR TITLE
Json simplification

### PR DESCRIPTION
**Pull request by david-i-berry**
Simplified JSON mapping files and removal of custom station metadata JSON files

### CODE REVIEW

File: csv2bufr/__init__.py [hunk 1]
```diff
@@ -44,6 +50,20 @@
 
 LOGGER = logging.getLogger(__name__)
 
+THISDIR = os.path.dirname(os.path.realpath(__file__))
+MAPPINGS = f"{THISDIR}{os.sep}resources{os.sep}mappings"
+
+_LATEST_=36
```
Comment (tomkralidis) [inline csv2bufr/__init__.py:56]: Is this the BUFR tables version?  Suggest `BUFR_VERSION`

File: csv2bufr/__init__.py [hunk 2]
```diff
@@ -44,6 +50,20 @@
...
+_ATTRIBUTES_ = ['code','units','scale','reference','width']
```
Comment (tomkralidis) [inline csv2bufr/__init__.py:57]: `ATTRIBUTES`

File: csv2bufr/__init__.py [hunk 3]
```diff
@@ -44,6 +50,20 @@
...
+_HEADERS_ = ["edition","masterTableNumber","bufrHeaderCentre",
```
Comment (tomkralidis) [inline csv2bufr/__init__.py:58]: `HEADERS`

File: csv2bufr/__init__.py [hunk 4]
```diff
@@ -196,258 +168,303 @@ def validate_value(key: str, value: Union[NUMBERS],
...
-
-    # ==============================
-    # Message now ready to be packed
-    # ==============================
-    try:
-        codes_set(bufr_msg, "pack", True)
-    except Exception as e:
-        LOGGER.error(f"error calling codes_set({bufr_msg}, 'pack', True): {e}")
-        raise e
+class bufr_message:
```
Comment (tomkralidis) [inline csv2bufr/__init__.py:171]: `class BUFRMessage:`

File: csv2bufr/__init__.py [hunk 5]
```diff
@@ -196,258 +168,303 @@ def validate_value(key: str, value: Union[NUMBERS],
...
-    message
-
-    :param bufr_msg: Integer used by eccodes to access message
-    :param object_: dictionary or list specifying what to extract
-                   from the BUFR message.
+        self.delayed_replications = delayed_replications
+        self.bufr = None
+
+
+    def set_element(self, key, value):
```
Comment (tomkralidis) [inline csv2bufr/__init__.py:206]: Add docstring

File: csv2bufr/__init__.py [hunk 6]
```diff
@@ -196,258 +168,303 @@ def validate_value(key: str, value: Union[NUMBERS],
...
-            args = extract(bufr_msg, object_["args"])
-            if None not in args:
-                result = object_["format"].format(*args)
-            else:
-                result = None
-        elif "eccodes_key" in object_:
-            result = codes_get(bufr_msg, object_["eccodes_key"])
-            if result in (CODES_MISSING_LONG, CODES_MISSING_DOUBLE):
-                result = None
+    def get_element(self, key:str):
```
Comment (tomkralidis) [inline csv2bufr/__init__.py:227]: add docstring

File: csv2bufr/__init__.py (hunk 5 above)
Comment (tomkralidis) [reply csv2bufr/__init__.py:206]: and function annotation

File: csv2bufr/__init__.py (hunk 6 above)
Comment (tomkralidis) [reply csv2bufr/__init__.py:227]: and function annotation

File: csv2bufr/__init__.py [hunk 7]
```diff
@@ -196,258 +168,303 @@ def validate_value(key: str, value: Union[NUMBERS],
...
+            tokens = key.split("->")
+            result = self.dict[tokens[0]][tokens[1]]
         else:
-            for k in object_:
-                object_[k] = extract(bufr_msg, object_[k])
+            result = self.dict[key]["value"]
+        return result
+
+
+    def as_bufr(self, force=False) -> bytes:
```
Comment (tomkralidis) [inline csv2bufr/__init__.py:237]: add docstring

File: csv2bufr/__init__.py [hunk 8]
```diff
@@ -196,258 +168,303 @@ def validate_value(key: str, value: Union[NUMBERS],
...
+            LOGGER.error(f"error writing to internal BytesIO object, {e}")
+            raise e
+        # =============================================
+        # Return BUFR message bytes
+        # =============================================
+        self.bufr = fh.read()
+        return self.bufr
+
+
+    def md5(self):
```
Comment (tomkralidis) [inline csv2bufr/__init__.py:298]: add docstring and function annotation

File: csv2bufr/__init__.py [hunk 9]
```diff
@@ -196,258 +168,303 @@ def validate_value(key: str, value: Union[NUMBERS],
...
+        return hashlib.md5(self.as_bufr()).hexdigest()
+
+    def as_geojson(self, identifier: str, template: dict) -> str:
```
Comment (tomkralidis) [inline csv2bufr/__init__.py:301]: add docstring

File: csv2bufr/__init__.py [hunk 10]
```diff
@@ -196,258 +168,303 @@ def validate_value(key: str, value: Union[NUMBERS],
...
+        result = self._extract(template)
+        result["id"] = identifier
+        result["properties"]["resultTime"] = datetime.now(timezone.utc).isoformat(timespec="seconds") #noqa
+        return json.dumps(result, indent=2)
+
+    def _extract(self, object_: Union[dict, list]) -> Union[dict, list]:
```
Comment (tomkralidis) [inline csv2bufr/__init__.py:307]: add docstring

File: csv2bufr/__init__.py [hunk 11]
```diff
@@ -196,258 +168,303 @@ def validate_value(key: str, value: Union[NUMBERS],
...
-                    # make sure column is in data_dict
-                    if (column not in data_dict):
-                        message = f"column '{column}' not found in data dictionary"  # noqa
-                        raise ValueError(message)
-                    value = data_dict[column]
-                    if value in MISSING:
+            result = object_
+        return result
+
+    def parse(self, data: str, metadata: dict, mappings: dict):
```
Comment (tomkralidis) [inline csv2bufr/__init__.py:333]: add return type in function annotation, add docstring

File: csv2bufr/__init__.py [hunk 12]
```diff
@@ -196,258 +168,303 @@ def validate_value(key: str, value: Union[NUMBERS],
...
+                        # ==================================================
+                        # at this point we should have the eccodes key and a
+                        # validated value to use, add to dict
+                        # ==================================================
+                        self.set_element(eccodes_key, value)
+                        LOGGER.debug(f"value {value} updated for element {element['eccodes_key']}")  # noqa
+            rows_read += 1
+
+
+    def datetime(self):
```
Comment (tomkralidis) [inline csv2bufr/__init__.py:432]: rename to `get_datetime`, add function annotation, add docstring

File: csv2bufr/__init__.py [hunk 13]
```diff
@@ -196,258 +168,303 @@ def validate_value(key: str, value: Union[NUMBERS],
...
+        "{:04d}-{:02d}-{:02d}T{:02d}:{:02d}:00+00:00".format(
+                                                             self.get_element("typicalYear"), #noqa
+                                                             self.get_element("typicalMonth"), #noqa
+                                                             self.get_element("typicalDay"), #noqa
+                                                             self.get_element("typicalHour"), #noqa
+                                                             self.get_element("typicalMinute")) #noqa
+
+
+def transform(data: str, metadata: dict, mappings: dict, template: dict = None)\
```
Comment (tomkralidis) [inline csv2bufr/__init__.py:441]: Add docstring

File: csv2bufr/cli.py [hunk 14]
```diff
@@ -28,7 +28,8 @@
 
 from csv2bufr import __version__
 from csv2bufr import transform as transform_csv
-from csv2bufr import bufr2geojson
+#from csv2bufr import bufr2geojson
```
Comment (tomkralidis) [inline csv2bufr/cli.py:31]: remove line

**Review by tomkralidis — CHANGES_REQUESTED** [review]

File: csv2bufr/__init__.py (hunk 1 above)
Comment (david-i-berry) [reply csv2bufr/__init__.py:56]: I've used BUFR_TABLE_VERSION.

File: csv2bufr/__init__.py (hunk 2 above)
Comment (david-i-berry) [reply csv2bufr/__init__.py:57]: changed

File: csv2bufr/__init__.py (hunk 3 above)
Comment (david-i-berry) [reply csv2bufr/__init__.py:58]: changed

File: csv2bufr/__init__.py (hunk 4 above)
Comment (david-i-berry) [reply csv2bufr/__init__.py:171]: changed

File: csv2bufr/__init__.py (hunk 5 above)
Comment (david-i-berry) [reply csv2bufr/__init__.py:206]: done

File: csv2bufr/__init__.py (hunk 10 above)
Comment (david-i-berry) [reply csv2bufr/__init__.py:307]: done

File: csv2bufr/__init__.py (hunk 11 above)
Comment (david-i-berry) [reply csv2bufr/__init__.py:333]: done

Comment (david-i-berry) [reply csv2bufr/__init__.py:333]: done

File: csv2bufr/__init__.py (hunk 12 above)
Comment (david-i-berry) [reply csv2bufr/__init__.py:432]: done

File: csv2bufr/__init__.py (hunk 13 above)
Comment (david-i-berry) [reply csv2bufr/__init__.py:441]: done

**Review by tomkralidis — APPROVED** [review]

**Pull request merged**